open_rarity/models/collection.py

```diff
@@ -231,5 +261,37 @@ def extract_collection_attributes(
 
         return collection_traits
 
+    def _derive_normalized_attributes_frequency_counts(
+        self,
+    ) -> dict[AttributeName, dict[AttributeValue, int]]:
+        """Derives and constructs attributes_frequency_counts based on
+        string attributes on tokens. Numeric or date attributes currently not
+        supported.
+
+        Returns
+        -------
+        dict[ AttributeName, dict[AttributeValue, int] ]
+            dictionary of attributes to the number of tokens in this collection
+            that has a specific value for every possible value for the given
+            attribute, by default None.
+        """
+        attrs_freq_counts: dict[
+            AttributeName, dict[AttributeValue, int]
+        ] = defaultdict(dict)
```

Comment:
You could do `defaultdict(lambda: defaultdict(int))` to get a nested default dict.

Reply:
i thought about it but then we would need to convert back to dict for each sub-dict and I thought tradeoff wasn't worth it. problem with defaultdict is that if you just read a key and it doesn't exist, it sets a defaultdict for it, which is not ideal, so we would have to convert